native/markdown/rules.react.js

```diff
@@ -6,12 +6,23 @@ import type { MarkdownStyles } from './styles';
 import * as React from 'react';
 import { Text, Linking } from 'react-native';
 import * as SimpleMarkdown from 'simple-markdown';
-import invariant from 'invariant';
 
 import { urlRegex } from 'lib/shared/markdown';
 
-export default function rules(styles: StyleSheetOf<MarkdownStyles>) {
-  return {
+type MarkdownRuleSpec = {|
+  +simpleMarkdownRules: SimpleMarkdown.ParserRules,
+  +emojiOnlyFactor: ?number,
```

Comment:
I wanted to make the `emojiOnly` decision part of the "rules", so I created a new structure that include both `simple-markdown` rules and `emojiOnlyFactor`